### PR TITLE
Add extension method for getting environment name

### PR DESCRIPTION
**Pull request by Praburaj**
Fixes: https://github.com/aspnet/Hosting/issues/100

@rustd @muratg

### CODE REVIEW

**Comment by muratg** [comment]
:shipit:

File: src/Microsoft.AspNet.Hosting.Interfaces/HostingEnvironmentExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        /// <summary>
+        /// Compares the current hosting environment name against the specified value.
+        /// </summary>
+        /// <param name="hostingEnvironment">An instance of <see cref="IHostingEnvironment"/> service.</param>
+        /// <param name="environmentName">Environment name to validate against.</param>
+        /// <returns>True if the specified name is same as the current environment.</returns>
+        public static bool IsEnvironment(
+            [NotNull]this IHostingEnvironment hostingEnvironment,
+            [NotNull]string environmentName)
```
Comment (Eilon) [inline src/Microsoft.AspNet.Hosting.Interfaces/HostingEnvironmentExtensions.cs]: Should this really throw on null?

Comment (Eilon) [inline src/Microsoft.AspNet.Hosting.Interfaces/HostingEnvironmentExtensions.cs]: (string.Equals() doesn't care about null, so the rest of the code would still work just fine)

Comment (Praburaj) [inline src/Microsoft.AspNet.Hosting.Interfaces/HostingEnvironmentExtensions.cs]: Is null a valid environment name? I think there is always a default in hosting layer. That's why I made it throw on null.

Comment (Eilon) [inline src/Microsoft.AspNet.Hosting.Interfaces/HostingEnvironmentExtensions.cs]: Yeah I'm thinking more of a mock/lazy implementation of the interface that happens to return null. Would we want that to fail?

Comment (Praburaj) [inline src/Microsoft.AspNet.Hosting.Interfaces/HostingEnvironmentExtensions.cs]: hmm. I think I can remove the null check on `environmentName`.

**Pull request merged**